src/main/java/baritone/command/defaults/ToMapCommand.java

```diff
@@ -0,0 +1,223 @@
+                    }
+                }
+            }
+        }
+        logDirect("Finished scanning");
+        int minX = Collections.min(topView.keySet());
+        int maxX = Collections.max(topView.keySet());
+        int minZ = Integer.MAX_VALUE;
+        int maxZ = Integer.MAX_VALUE;
+        for (Map<Integer, String> zId : topView.values()) {
+            int minZForX = Collections.min(zId.keySet());
+            if (minZForX < minZ)
+                minZ = minZForX;
+            int maxZForX = Collections.max(zId.keySet());
+            if (maxZForX < maxZ)
+                maxZ = maxZForX;
+        }
+        int offsetX = -minX;
+        int offsetZ = -minZ;
+        BufferedImage result = new BufferedImage(maxX + offsetX, maxZ + offsetZ, BufferedImage.TYPE_INT_ARGB);
+        for (int x = minX; x < maxX; x++) {
+            for (int z = minZ; z < maxZ; z++) {
+                String blockId = null;
+                try {
+                    blockId = topView.get(x).containsKey(z) ? topView.get(x).get(z) : "air";
+                    if (!blockId.equals("air")) {
+                        BufferedImage textureAtCoord = getTextureForBlockId(blockId);
+                        Color c = averageColor(textureAtCoord, blockId);
+                        result.setRGB(x, z, c.getRGB());
+                    } else {
+                        result.setRGB(x, z, Color.TRANSLUCENT);
+                    }
+                } catch (IOException e) {
+                    System.out.println("Error setting color for " + blockId);
+                    e.printStackTrace();
+                }
+            }
+        }
+        logDirect("Finished coloring");
+        try {
+            ImageIO.write(result, "png", new File("/home/nexor/output.png"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public Color averageColor(BufferedImage bi, String id) {
+        if (!cachedColors.containsKey(id)) {
+//            System.out.println(id);
+            int x1 = 16;
+            int y1 = 16;
+            long sumr = 0, sumg = 0, sumb = 0;
+            for (int x = 0; x < x1; x++) {
+                for (int y = 0; y < y1; y++) {
+                    Color pixel = new Color(bi.getRGB(x, y));
+                    sumr += pixel.getRed();
+                    sumg += pixel.getGreen();
+                    sumb += pixel.getBlue();
+                }
+            }
+            int num = 16 * 16;
+            try {
+                cachedColors.put(id, new Color(sumr / num, sumg / num, sumb / num));
+            } catch (Exception e) {
+                cachedColors.put(id, new Color(Color.TRANSLUCENT));
+            }
+        }
+        return cachedColors.get(id);
+    }
+
+    private static BufferedImage getTextureForBlockId(String blockId) throws IOException {
+        String base = "/home/nexor/.minecraft/versions/1.15.2-OptiFine_HD_U_G5_pre1/1.15.2-OptiFine_HD_U_G5_pre1/assets/minecraft/textures/block/";
```

Comment:
another hardcoded path

Reply:
How can we fix this?

Reply:
Don't know if https://github.com/cabaletta/baritone/pull/2257#issuecomment-743911392 is enough here but I think it should be possible to find the current version's assets

Reply:
The latter would be best, since not everyone has an unzipped jar lying around in their versions directory lol